test/integration/query-interface/getForeignKeyReferencesForTable.test.js

```diff
@@ -41,5 +44,86 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
       expect(refs[0]).deep.include.all(expectedObject);
 
     });
+
+    it('should be able to provide existing foreign keys when multiple associations exist', async function () {
+      const Author    = this.sequelize.define('Author',    { name: DataTypes.STRING });
+      const Book      = this.sequelize.define('Book',      { title: DataTypes.STRING });
+      const Publisher = this.sequelize.define('Publisher', { name: DataTypes.STRING });
+      const Critic    = this.sequelize.define('Critic',    { name: DataTypes.STRING });
+
+      Author.hasMany(Book);
+      Publisher.hasMany(Book);
+      Book.belongsTo(Publisher);
+      Book.belongsTo(Author);
+      Book.belongsToMany(Critic, { through: 'Reviews' });
+      Critic.belongsToMany(Book, { through: 'Reviews' });
+
+      const options = { force: true };
+
+      await Author.sync(options);
+      await Publisher.sync(options);
+      await Book.sync(options);
+      await Critic.sync(options);
```

Comment:
any reason for not using this instead?

```suggestion
      await this.sequelize.sync({ force: true });
```